python/vyos/ifconfig/interface.py

```diff
@@ -977,25 +977,55 @@ def del_tc_qdisc(self,dev,kind,handle):
             old_handle = rule['handle']
             old_kind = rule['kind']
             if old_dev == dev and old_handle == handle and old_kind == kind:
-                delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} {kind}'
-                self._cmd(delete_tc_cmd)
+                if 'root' in rule and rule['root']:
+                    delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} root {kind}'
+                    self._cmd(delete_tc_cmd)
+                else:
+                    delete_tc_cmd = f'tc  qdisc del dev {dev} handle {handle} {kind}'
+                    self._cmd(delete_tc_cmd)
 
 
 
-    def apply_mirror(self,config):
-        ifname = config['ifname']
-
+    def apply_mirror(self):
+        # Please refer to the document for details
+        # https://man7.org/linux/man-pages/man8/tc.8.html
+        # https://man7.org/linux/man-pages/man8/tc-mirred.8.html
+        ifname = self._config['ifname']
         # Remove existing mirroring rules
         self.del_tc_qdisc(ifname,'ingress','ffff:')
-
+        self.del_tc_qdisc(ifname,'prio','1:')
+        
         # Setting up packet mirroring
-        mirror = dict_search('mirror', config)
+        mirror = dict_search('mirror', self._config)
         if mirror:
-            for interface in mirror:
+            ingress = False
+            egress = False
+            for interface, interface_conf in mirror.items():
+                if 'ingress' in interface_conf:
+                    ingress = True
+                
+                if 'egress' in interface_conf:
+                    egress = True
+            
+            if ingress:
+                # Mirror ingress traffic
                 mirror_cmd = f'tc qdisc add dev {ifname} handle ffff: ingress'
                 self._cmd(mirror_cmd)
-                mirror_cmd = f'tc filter add dev {ifname} parent ffff: protocol all prio 10 u32 match u32 0 0 flowid 1:1 action mirred egress mirror dev {interface}'
+            
+            if egress:
+                # Mirror egress traffic
+                mirror_cmd = f'tc qdisc add dev {ifname} handle 1: root prio'
                 self._cmd(mirror_cmd)
+            
+            for interface, interface_conf in mirror.items():
+                if 'ingress' in interface_conf:
```

Comment:
Why not combine this with the above code? Then you will be left with one control path only.